library/core/src/num/nonzero.rs

```diff
@@ -287,6 +287,204 @@ nonzero_integers_div! {
+                /// Return [`None`] on overflow.
+                ///
+                /// # Examples
+                ///
+                /// ```
+                /// # #![feature(nonzero_add)]
+                #[doc = concat!("# use std::num::", stringify!($Ty), ";")]
+                ///
+                #[doc = concat!("let one = ", stringify!($Ty), "::new(1).unwrap();")]
+                #[doc = concat!("let two = ", stringify!($Ty), "::new(2).unwrap();")]
+                #[doc = concat!("let max = ", stringify!($Ty), "::new(",
+                                stringify!($Int), "::MAX).unwrap();")]
+                /// assert_eq!(Some(two), one.checked_add(1));
+                /// assert_eq!(None, max.checked_add(1));
+                /// ```
+                #[unstable(feature = "nonzero_add", issue = "none")]
+                #[inline]
+                pub fn checked_add(self, other: $Int) -> Option<$Ty> {
+                    self.get().checked_add(other).map(|result| {
+                        // SAFETY: $Int::checked_add returns None on overflow
+                        // so the result cannot be zero.
+                        unsafe { $Ty::new_unchecked(result) }
+                    })
+                }
+
+                /// Add a positive integer to a non-zero value.
+                #[doc = concat!("Return [`", stringify!($Int), "::MAX`] on overflow.")]
+                ///
+                /// # Examples
+                ///
+                /// ```
+                /// # #![feature(nonzero_add)]
+                #[doc = concat!("# use std::num::", stringify!($Ty), ";")]
+                #[doc = concat!("let one = ", stringify!($Ty), "::new(1).unwrap();")]
+                #[doc = concat!("let two = ", stringify!($Ty), "::new(2).unwrap();")]
+                #[doc = concat!("let max = ", stringify!($Ty), "::new(",
+                                stringify!($Int), "::MAX).unwrap();")]
+                /// assert_eq!(two, one.saturating_add(1));
+                /// assert_eq!(max, max.saturating_add(1));
+                /// ```
+                #[unstable(feature = "nonzero_add", issue = "none")]
+                #[inline]
+                pub fn saturating_add(self, other: $Int) -> $Ty {
+                    // SAFETY: $Int::saturating_add returns $Int::MAX on overflow
+                    // so the result cannot be zero.
+                    unsafe { $Ty::new_unchecked(self.get().saturating_add(other)) }
+                }
+            }
+        )+
+    }
+}
+
+nonzero_integers_add! {
+    NonZeroU8(u8);
+    NonZeroU16(u16);
+    NonZeroU32(u32);
+    NonZeroU64(u64);
+    NonZeroU128(u128);
+    NonZeroUsize(usize);
+}
+
+macro_rules! nonzero_integers_mul {
+    ( $( $Ty: ident($Int: ty); )+ ) => {
+        $(
+            impl $Ty {
+                /// Multiply two non-zero integers together.
+                /// Return [`None`] on overflow or if `other` is zero.
```

Comment:
> ``or if `other` is zero.``

`other` is NonZero, right?

Reply:
Wops, very true, it used not to be thus the mistake :)